File: cpp/089_GrayCode.cpp
//Backtracking

//bit
class Solution {
public:
    vector<int> grayCode(int n) {
        int len = 1 << n;
        vector<int> ret;
        for(int i = 0; i < len; ++i){
            ret.push_back(i ^ (i >> 1));
        }
        return ret;
    }
};

File: js/089_GrayCode.js
//Backtracking

//bit
/**
 * @param {number} n
 * @return {number[]}
 */
var grayCode = function(n) {
    var len = 1 << n;
    var ret = [];
    for(var i = 0; i < len; ++i){
        ret.push( i ^ (i >> 1));
    }
    return ret;
};